cmd/spiffe-helper/config.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"io/ioutil"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spiffe-helper/pkg/sidecar"
+)
+
+// ParseConfig parses the given HCL file into a SidecarConfig struct
+func ParseConfig(file string) (Config *sidecar.Config, err error) {
+	sidecarConfig := &sidecar.Config{
+		AgentAddress:       `hcl:"agentAddress"`,
```

Comment:
the fields are being initialized to hcl struct tags? I think we probably just want the following:
```suggestion
		sidecarConfig := new(sidecar.Config)
```
Those struct tags belong on the sidecar.Config fields...

Reply:
I moved the hcl stuff out here because we use this as a package and don't load config from a file. I thought it made more sense to have the struct tags closer to where the config is read from a file. Happy to put it back where it was if you think thats a better approach.

Reply:
I see. In that case, we probably want to duplicate the struct into the `main` package, attach the `hcl` tags and initialize `sidecar.Config` with that struct. It does make it so that newly added config needs to be threaded through one more spot, but that makes supporting a new sidecar configuration from `spiffe-helper` more explicit (since you explicitly have to add a new field to the "hcl" struct, update documentation, etc.)

In either case, these field assignments aren't doing anything useful other than initializing the struct field values with garbage.

Reply:
I moved it back to the sidecar.go and that works fine. If the tags aren't used they are ignored.